File: create_notifications_table.mjs
// Create the notifications table via Supabase Management API (POST /v1/projects/{ref}/sql)
// The service role key used so far is the database JWT key — try the Management API;
// if rejected, fall back to creating the table lazily through the app itself is not possible.
import { createClient } from '@supabase/supabase-js';

const REF = 'uqtirisxgqmhxupncink';
const SERVICE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InVxdGlyaXN4Z3FtaHh1cG5jaW5rIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4NjY1NjI0MywiZXhwIjoyMTAyMjMyMjQzfQ.patjJ_GGmXM2xrgLikXEYeHz6WZzDZPwH8vAyatB438';

const url = `https://api.supabase.com/v1/projects/${REF}/sql`;
const sql = `
CREATE TABLE IF NOT EXISTS public.notifications (
  id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
  user_id INTEGER NULL, -- NULL = broadcast to all users
  title TEXT NOT NULL,
  body TEXT NOT NULL,
  is_broadcast BOOLEAN NOT NULL DEFAULT FALSE,
  created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
);
CREATE INDEX IF NOT EXISTS idx_notifications_user ON public.notifications (user_id);
`;

const res = await fetch(url, {
  method: 'POST',
  headers: {
    'Authorization': `Bearer ${SERVICE_KEY}`,
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({ query: sql }),
});
const txt = await res.text();
console.log('STATUS', res.status);
console.log(txt.slice(0, 500));

File: create-notifications-table.py
import json, urllib.request

URL = "https://uqtirisxgqmhxupncink.supabase.co/rest/v1/rpc/exec_sql"
KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InVxdGlyaXN4Z3FtaHh1cG5jaW5rIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc4NjY1NjI0MywiZXhwIjoyMTAyMjMyMjQzfQ.patjJ_GGmXM2xrgLikXEYeHz6WZzDZPwH8vAyatB438"

# Try the edge-function SQL RPC first; fall back to direct DDL via /rest/v1 (not supported), so use RPC.
SQL = """
CREATE TABLE IF NOT EXISTS public.notifications (
  id bigserial PRIMARY KEY,
  user_id bigint NULL,
  is_broadcast boolean NOT NULL DEFAULT false,
  title text NOT NULL,
  body text NOT NULL DEFAULT '',
  kind text NOT NULL DEFAULT 'broadcast',
  is_read boolean NOT NULL DEFAULT false,
  created_at timestamptz NOT NULL DEFAULT now()
);
"""

body = json.dumps({"sql": SQL}).encode()
req = urllib.request.Request(URL, data=body, method="POST", headers={
    "apikey": KEY, "Authorization": f"Bearer {KEY}",
    "Content-Type": "application/json", "Prefer": "return=representation"})
try:
    with urllib.request.urlopen(req, timeout=30) as r:
        print("OK", r.status, r.read().decode()[:300])
except Exception as e:
    print("RPC failed:", e)
    # Fallback: use supabase-js style via management API is not possible; print fallback note
    print("FALLBACK: need another way (e.g. create via JS client postgrest insert to test existence)")
